crates/re_web_viewer_server/src/lib.rs

```diff
@@ -191,73 +89,186 @@ impl FromStr for WebViewerServerPort {
+                    continue;
+                }
+            };
+
+            if let Err(err) = self.send_response(request) {
+                re_log::error!("Failed to send http response: {err}");
+            }
+        }
+    }
+
+    #[cfg(feature = "analytics")]
+    fn on_serve_wasm(&self) {
+        if let Some(analytics) = &self.analytics {
+            analytics.record(re_analytics::event::ServeWasm);
+        }
+    }
+
+    #[cfg(feature = "__ci")]
+    #[allow(clippy::needless_pass_by_value)]
+    fn send_response(&self, _request: tiny_http::Request) -> Result<(), std::io::Error> {
+        if false {
+            self.on_serve_wasm(); // to silence warning about the function being unused
+        }
+
+        // panic! is not enough in hyper (since it uses catch_unwind) - that only kills this thread. We want to quit.
+        eprintln!("web_server compiled with '__ci' feature (or `--all-features`). DON'T DO THAT! It's only for the CI!");
+        std::process::abort();
+    }
+
+    #[cfg(not(feature = "__ci"))]
+    fn send_response(&self, request: tiny_http::Request) -> Result<(), std::io::Error> {
+        // Strip arguments from url so we get the actual path.
+        let url = request.url();
+        let path = url.split('?').next().unwrap_or(url);
+
+        let (mime, bytes) = match path {
+            "/" | "/index.html" => ("text/html", data::INDEX_HTML),
+            "/favicon.svg" => ("image/svg+xml", data::FAVICON),
+            "/favicon.ico" => {
+                // We don't have a favicon.ico.
+                return request.respond(tiny_http::Response::empty(404));
+            }
+            "/sw.js" => ("text/javascript", data::SW_JS),
+            "/re_viewer.js" => ("text/javascript", data::VIEWER_JS),
+            "/re_viewer_bg.wasm" => {
+                #[cfg(feature = "analytics")]
+                self.on_serve_wasm();
+                ("application/wasm", data::VIEWER_WASM)
+            }
+            _ => {
+                re_log::warn!("404 path: {}", path);
+                return request.respond(tiny_http::Response::empty(404));
+            }
+        };
+
+        let mut response = tiny_http::Response::from_data(bytes).with_header(
+            tiny_http::Header::from_str(&format!("Content-Type: {mime}"))
+                // Both `mime` and the header are hardcoded, so shouldn't be able to fail depending on user input.
+                .expect("Invalid http header"),
+        );
+
+        // The wasm files are pretty large, so they'll be sent chunked (ideally we'd gzip them...).
+        // Unfortunately `Transfer-Encoding: chunked` means that no size is transmitted.
+        // We work around this by adding a custom header with the size that web_viewer/index.html understands.
+        if path.ends_with(".wasm") {
+            if let Ok(header) =
+                tiny_http::Header::from_str(&format!("rerun-wasm-size: {}", bytes.len()))
+            {
+                response.add_header(header);
```

Comment:
I suggest just generalize this as `rerun-final-lenth: {}` or something and send it for all responses, because why not 🤷